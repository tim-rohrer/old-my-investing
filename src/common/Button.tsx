import React from 'react'

type ButtonProps = {
    onClick: any,
    className?: string,
    children: any
}

function Button({onClick, 
    className, 
    children}: ButtonProps) {

    return (
        <button
            onClick={onClick}
            className={className}
            type="button"
        >
            {children}
        </button>
    );
}

export default Button;