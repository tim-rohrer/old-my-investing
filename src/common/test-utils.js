// test-utils.js
import React from "react";
// import { render as rtlRender } from '@testing-library/react';
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { createStore } from "redux";
import { Provider } from "react-redux";
import {
  initialStateOriginal as reducerInitialState,
  reducer,
  initialStateOriginal,
} from "../app/store";

// const store = configureStore({ reducer });

export function renderWithRedux(
  ui,
  {
    initialState = initialStateOriginal,
    store = configureStore({ reducer }),
  } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}
// export function customRender(
//   ui,
//   {
//     initialState: RootState = reducerInitialState,
//     rtlStore,
//     ...renderOptions
//   } = {}
// ) {
//   function Wrapper({ children }) {
//     const rtlStore = configureStore({ reducer });
//     return <Provider store={rtlStore}>{children}</Provider>;
//   }
//   return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
// }

// re-export everything
export * from "@testing-library/react";

// override render method
// export { customRender as render };
