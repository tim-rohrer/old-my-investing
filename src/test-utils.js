// test-utils.js
import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
// import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { reducer } from './app/store'
// import { initialState as reducerInitialState, reducer } from './reducer'
const reducerInitialState = {}

function render(
  ui,
  {
    initialState: RootState = reducerInitialState,
    rtlStore,
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const rtlStore = configureStore({ reducer })
    return <Provider store={rtlStore}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'

// override render method
export { render }