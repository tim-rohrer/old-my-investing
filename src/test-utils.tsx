// test-utils.tsx
import * as React from "react"
import { Provider } from "react-redux"
import { createStore } from "redux"

import { queries, Queries } from "@testing-library/dom"
import { render as rtlRender, RenderOptions, RenderResult } from "@testing-library/react"

import { rootReducer, State, Store } from "./app/store"

type ExtraOptions = {
  initialState?: State;
  store?: Store;
};

const render = <
  Q extends Queries = typeof queries,
  C extends Element | DocumentFragment = HTMLElement
>(
  ui: React.ReactElement,
  options: ExtraOptions & RenderOptions<Q, C> = {} //:
): RenderResult<Q, C> & { store: Store } => {
  const {
    initialState = undefined, // reducerInitialState,
    store = createStore(rootReducer, initialState),
    ...renderOptions
  } = options;

  type Props = {
    children?: React.ReactNode
  }

  const Wrapper = ({ children }: Props) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return {
    ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }),
    store,
  };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
