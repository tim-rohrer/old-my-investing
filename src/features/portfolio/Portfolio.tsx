import React from "react"
import { selectPortfolioLoadedStatus } from "./portfolioSlice"
import { useSelector } from "react-redux"
import Button from "../../common/Button"

export const callAlert = () => {
    alert("Here!")
}

function Portfolio() {

    let isPortfolioLoaded = useSelector(selectPortfolioLoadedStatus)

    return (
        <div>
            <h2>Hello!</h2>
            <span>The portfolio
                {
                    isPortfolioLoaded
                    ?
                    <p> is loaded.</p>
                    :
                    <p>is not yet loaded.</p>
                }
            </span>
            <Button onClick={() => callAlert()}>Alert!</Button>
        </div>
    )

}

export default Portfolio