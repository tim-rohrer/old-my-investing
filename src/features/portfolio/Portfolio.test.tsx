import React from "react";
import { render, cleanup } from "../../test-utils";
import Portfolio from "./Portfolio";
import Button from "../../common/Button";

afterEach(cleanup);

describe("Portfolio", () => {
  it("renders a snapshot", () => {
    const { container } = render(<Portfolio />);
    expect(container).toMatchSnapshot();
  });
  // it("fires event", () => {
  //   const { container, asFragment } = render(
  //     <Button onClick={() => callAlert()}>Alert!</Button>
  //   );
  //   fireEvent.click(container);
  //   // expec
  //   // const { container } = render(<Portfolio />)
  //   // const node = getByDisplayValue(container, 'Alert!')
  //   // const myEvent = createEvent.click(node, { button: 1 })
  //   // fireEvent(node, myEvent)
  // });
});
