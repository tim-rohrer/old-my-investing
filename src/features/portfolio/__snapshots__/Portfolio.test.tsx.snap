// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio renders a snapshot 1`] = `
<div>
  <div>
    <h2>
      Hello!
    </h2>
    <span>
      The portfolio
      <p>
        is not yet loaded.
      </p>
    </span>
    <button
      type="button"
    >
      Alert!
    </button>
  </div>
</div>
`;
